indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,548 @@
+                    $this.trigger('input.typeahead');
+                    venueJustChanged = false;
+                }
+                $('.ui-dialog-content').css('overflow', 'inherit');  // Make sure the results dropdown are displayed
+                                                                     // above the dialog.
+                toggleOpenResultsList($this);
+                latestUsedField = $(this);
+            })
+            .on('keydown', function(e) {
+                var $this = $(this);
+                if (resultsWereOpen) {
+                    if (e.keyCode == K.TAB) {
+                        if ($this.val()) {
+                            // Treat tab as the return key
+                            var returnKey = jQuery.Event("keydown");
+                            returnKey.which = K.ENTER;
+                            returnKey.keyCode = K.ENTER;
+                            $this.trigger(returnKey);
+                        } else {
+                            clearField($this);
+                        }
+                    } else if (e.keyCode == K.ENTER) {
+                        e.preventDefault();
+                        if ($this.val()) {
+                            if (latestSelectedOption.value == '' || latestSelectedOption.field == $this.attr('id') && latestSelectedOption.value != $this.val()) {
+                                $this.trigger('typeahead:click-on-custom-option', {item_name: $this.val()});
+                            }
+                        } else {
+                            clearField($this);
+                        }
+                    } else if (e.keyCode == K.ESCAPE) {
+                        e.preventDefault();
+                        resetSelectedOption();
+                    } else if (~[K.UP, K.DOWN].indexOf(e.keyCode)) {
+                        scrollResultsList($this);
+                        $this.trigger('focus.typeahead');
+                    }
+                }
+            })
+            .on('click', function() {
+                /* Since focus remains on the field after selection, we need special handling for the click event in
+                order to reopen the results list */
+                Typeahead["#" + $(this).attr('id')].showLayout();
+                $(this).trigger('input');
+            });
+
+            $('.toggle-results-list').on('click', function() {
+                var container = $(this).parent().parent();
+                var input = container.find('.i-location-input-field').eq(0);
+                input.trigger(container.hasClass('result') ? 'typeahead:close-results-list' : 'focus.typeahead');
+            });
+
+            address.on('keyup', function() {
+                hiddenData.address = address.val();
+                field.val(JSON.stringify(hiddenData)).trigger('change');
+            });
+
+            $('.typeahead-field .highlighter').on('click', function() {
```

Comment:
too generic. always restrict stuff in widgets to that specific field's widget using the field id